zion/src/main/scala/edu/uci/ics/cloudberry/zion/actor/DataStoreManager.scala

```diff
@@ -161,12 +161,28 @@ class DataStoreManager(metaDataset: String,
     val dropTableName = dropTable.dataset
 
     if(metaData.contains(dropTableName)){
-      metaData.remove(dropTableName)
+      val metaRecordFilter = FilterStatement(DataSetInfo.MetaSchema.fieldMap("name"), None, Relation.matches, Seq(dropTableName))
+      metaActor ! DeleteRecord(metaDataset, Seq(metaRecordFilter))
+
+      metaData.foreach( dataSchema =>
+        dataSchema._2.createQueryOpt match {
+          case Some(query) =>
+            if(query.dataset == dropTableName){
+              metaActor ! DropView(dataSchema._1)
+              val viewRecordFilter = FilterStatement(DataSetInfo.MetaSchema.fieldMap("name"), None, Relation.matches, Seq(dataSchema._1))
+              metaActor ! DeleteRecord(metaDataset, Seq(viewRecordFilter))
```

Comment:
OK. What is the logic when `ViewActor` and `OriginActor` are created? From the code, does it mean that one dataset only have one `ViewActor` or `OriginalActor` with the name "data-(dataset name)"? My understand is that when a `Query` or `AppendView` request comes, an actor named "data-(dataset name)" will be called if existed, or created according to if `createQueryOpt` is empty or not. Am I right?